### PR TITLE
Add data source get policies

### PR DESCRIPTION
**Pull request by bardielle**

### CODE REVIEW

File: provider/ocm_policies_data_source.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		"sts_instance_controlplane_permission_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+	})
+}
+
+func operatorRolePoliciesNames() tfsdk.NestedAttributes {
+	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+		"openshift_cloud_credential_operator_cloud_credential_operator_iam_ro_creds_policy": {
```
Comment (gdbranco) [inline provider/ocm_policies_data_source.go:96]: Could use the consts you setup above
https://github.com/terraform-redhat/terraform-provider-ocm/pull/36/files#diff-229c93a91ec2ce9ad63f6c459ed2a928c23d23bed45534f77ddc80387e7db576R31

File: provider/ocm_policies_data_source.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+)
+
+const (
+	// Policy IDs from type operator roles
+	CloudCred       = "openshift_cloud_credential_operator_cloud_credential_operator_iam_ro_creds_policy"
```
Comment (gdbranco) [inline provider/ocm_policies_data_source.go:31]: I feel like in ROSA we check it with some sort of regex or something similar
I just have a feeling that having these as consts will complicate things when adding new roles/policies

File: provider/ocm_policies_data_source.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	CloudNetwork    = "openshift_cloud_network_config_controller_cloud_credentials_policy"
+	ClusterCSI      = "openshift_cluster_csi_drivers_ebs_cloud_credentials_policy"
+	ImageRegistry   = "openshift_image_registry_installer_cloud_credentials_policy"
+	IngressOperator = "openshift_ingress_operator_cloud_credentials_policy"
+	MachineAPI      = "openshift_machine_api_aws_cloud_credentials_policy"
+
+	// Policy IDs from type account roles
+	Installer            = "sts_installer_permission_policy"
```
Comment (gdbranco) [inline provider/ocm_policies_data_source.go:39]: Hypershift clusters have a separate set of account and operator roles, I'd like to tag Ori on this as he's handling the work for it

File: provider/ocm_policies_data_source.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	response *tfsdk.ReadDataSourceResponse) {
+	// Get the state:
+	state := &OcmPoliciesState{}
+	diags := request.Config.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	policiesResponse, err := t.awsInquiries.STSPolicies().List().Send()
```
Comment (gdbranco) [inline provider/ocm_policies_data_source.go:149]: This request get the actual policy document, but there's also
`stsCredentialResponse, err := c.ocm.ClustersMgmt().
		V1().
		AWSInquiries().
		STSCredentialRequests().
		List().
		Parameter("is_hypershift", isHypershift).
		Send()`
		
Which I think could be used to remove the constant for the policy names

File: provider/ocm_policies_data_source.go (hunk 3 above)
Comment (bardielle) [reply provider/ocm_policies_data_source.go:39]: it sounds like for hypershift should separate the creation

Comment (oriAdler) [reply provider/ocm_policies_data_source.go:39]: @gdbranco correct, for Hypershift we're going to have only three account roles: installer, support, and instance worker.

Also, the default flow for Hypershfit will use managed policies, and the policies ARNs should be sufficient.

File: provider/ocm_policies_data_source.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+type OcmPoliciesDataSource struct {
+	logger       logging.Logger
+	awsInquiries *cmv1.AWSInquiriesClient
+}
+
+func (t *OcmPoliciesDataSourceType) GetSchema(ctx context.Context) (result tfsdk.Schema,
+	diags diag.Diagnostics) {
+	result = tfsdk.Schema{
+		Description: "List of rosa operator role for a specific cluster.",
```
Comment (oriAdler) [inline provider/ocm_policies_data_source.go:56]:
```suggestion
		Description: "List of rosa operator roles and account roles for a specific cluster.",
```

File: provider/ocm_policies_data_source.go [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+		},
+	})
+}
+
+func (t *OcmPoliciesDataSourceType) NewDataSource(ctx context.Context,
+	p tfsdk.Provider) (result tfsdk.DataSource, diags diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation:
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
```
Comment (oriAdler) [inline provider/ocm_policies_data_source.go:128]: Under `/aws_inquiries/` we have multiple endpoints, but none returns a collection of clusters.
Not sure the comment is accurate.

File: provider/ocm_policies_data_source.go (hunk 3 above)
Comment (bardielle) [reply provider/ocm_policies_data_source.go:39]: So can we resolve that conversation? @oriAdler @gdbranco ?

File: provider/ocm_policies_data_source.go (hunk 6 above)
Comment (bardielle) [reply provider/ocm_policies_data_source.go:128]: Fix it to `collection of aws inquiries` thanks

**Comment by oriAdler** [comment]
LGTM

**Review by gdbranco — APPROVED** [review]
LGTM

**Pull request merged**